pkg/ipcache/metadata_test.go

```diff
@@ -236,6 +236,63 @@ func TestInjectLabels(t *testing.T) {
 	option.Config.EnableIPv6 = true
 }
 
+// Test that when multiple IPs have the `resolved:host` label, we correctly
+// aggregate all labels *and* update the selector cache correctly.
+// This reproduces GH-28259.
+func TestUpdateLocalNode(t *testing.T) {
+	cancel := setupTest(t)
+	defer cancel()
+
+	ctx := context.Background()
+
+	hostid := identity.ReservedIdentityHost
+
+	bothLabels := labels.Labels{}
+	bothLabels.MergeLabels(labels.LabelHost)
+	bothLabels.MergeLabels(labels.LabelKubeAPIServer)
+
+	selectorCacheHas := func(lbls labels.Labels) {
+		t.Helper()
+		id := PolicyHandler.identities[hostid]
+		assert.NotNil(t, id)
+
+		assert.True(t, id.Equals(lbls.LabelArray()), "labels should equal: %v -- %v", id, lbls.LabelArray())
+	}
+
+	// Mark .4 as local host
+	IPIdentityCache.metadata.upsertLocked(inClusterPrefix, source.Local, "node-uid", labels.LabelHost)
+	remaining, err := IPIdentityCache.InjectLabels(ctx, []netip.Prefix{inClusterPrefix})
+	assert.NoError(t, err)
+	assert.Len(t, remaining, 0)
+	assert.Equal(t, IPIdentityCache.ipToIdentityCache["10.0.0.4/32"].ID, hostid)
+	selectorCacheHas(labels.LabelHost)
+
+	// Mark .4 as kube-apiserver
+	IPIdentityCache.metadata.upsertLocked(inClusterPrefix, source.Local, "kube-uid", labels.LabelKubeAPIServer)
```

Comment:
This should actually be

```suggestion
	IPIdentityCache.metadata.upsertLocked(inClusterPrefix, source.KubeAPIServer, "kube-uid", labels.LabelKubeAPIServer)
```

but I don't think there's a functional impact here.

Reply:
Keeping the sources the same makes the bug much more likely. So it is a good test-case for the specific bug, even if it doesn't mirror current usage exactly.

Reply:
> Keeping the sources the same makes the bug much more likely

The only cases of labels attaching to the host identity is either kube-apiserver or CIDR labels in the case of node CIDR policy. AFAICT, CIDR labels upserted as metadata into the ipcache from the nodemanager with the same `Local` source because the node being handled in the local node.

In this test case, we are testing the association of the kube-apiserver label which always comes with the `KubeAPIServer` source. If we were testing the association of CIDR labels, I'd agree.

Anyway, I think this is a very minor thing because there's no functional impact. I was more calling it out from the perspective of a future reader potentially getting confused and thinking there's a bug in the test.

Reply:
Added a comment to this effect -- that the source is normally different, but it's the same in the test case to maximize the chance of triggering the bug.